src/status_im/biometric_auth/core.cljs

```diff
@@ -0,0 +1,62 @@
+(ns status-im.biometric-auth.core
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.platform :as platform]
+            [status-im.i18n :as i18n]
+            [status-im.utils.fx :as fx]
+            [status-im.ui.components.colors :as colors]
+            [status-im.react-native.js-dependencies :as rn]))
+
+;; biometric auth config
+;; https://github.com/naoufal/react-native-touch-id#authenticatereason-config
+(def authenticate-options
+  (clj->js (merge
+            {:unifiedErrors true
+             :passcodeFallback false}
+            (when platform/ios?
+              {})
+            (when platform/android?
+              {:title "Authentication Required"
+               :imageColor colors/blue
+               :imageErrorColor colors/red
+               :sensorDescription "Touch sensor"
+               :sensorErrorDescription "Failed"
+               :cancelText (i18n/label :cancel)}))))
+
+(defn- get-error-message
+  "must return an error message for the user"
+  [touchid-error-obj]
+  (let [code (aget touchid-error-obj "code")]
+    (cond
+      ;; no message if user canceled or falled back to password
+      (= code "USER_CANCELED") nil
+      (= code "USER_FALLBACK") nil
+      ;; add here more specific errors if needed
+      ;; https://github.com/naoufal/react-native-touch-id#unified-errors
+      :else (i18n/label :biometric-auth-error {:code code}))))
+
+(defn get-supported [callback]
+  (if platform/desktop?
+    (callback false)
+    (-> (.isSupported rn/touchid)
```

Comment:
I'd love to see a black/whitelist of devices because generally speaking biometric auth on some android phones is garbage: https://www.zdnet.com/article/nokia-9-buggy-update-lets-anyone-bypass-fingerprint-scanner-with-a-pack-of-gum/

Reply:
very good point